src/compiler/checker.ts

```diff
@@ -12073,7 +12073,21 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
 
     function addInheritedMembers(symbols: SymbolTable, baseSymbols: Symbol[]) {
         for (const s of baseSymbols) {
-            if (!symbols.has(s.escapedName) && !isStaticPrivateIdentifierProperty(s)) {
+            if (isStaticPrivateIdentifierProperty(s)) {
+                continue;
+            }
+            const derived = symbols.get(s.escapedName);
+            if (derived) {
+                // non-constructor/static-block assignment declarations are ignored here; they're not treated as overrides
+                if (derived.valueDeclaration
+                    && isBinaryExpression(derived.valueDeclaration)
+                    && !isConstructorDeclaredProperty(derived)
+                    && !findAncestor(derived.valueDeclaration, n => isClassLike(n) ? "quit" : isClassStaticBlockDeclaration(n))) {

```

Comment:
Could you use this?

```suggestion
                    && !getContainingClassStaticBlock(derived.valueDeclaration))
```

The semantics are slightly different because it quits on function-like nodes.

Reply:
Yep, quitting on function-like nodes makes sense for function declarations and expressions since they bind `this`. Thanks for pointing this out.